aspnetcore/fundamentals/servers/kestrel.md

```diff
@@ -741,69 +711,33 @@ The following example permits HTTP/1.1 and HTTP/2 connections on port 8000. Conn
 });
 ```
 
-Optionally use Connection Middleware to filter TLS handshakes on a per-connection basis for specific ciphers:
+Use `CipherSuitesPolicy` to filter TLS handshakes on a per-connection basis for specific ciphers if required:
 
 ```csharp
-// using System.Net;
-// using Microsoft.AspNetCore.Connections;
+// using System.Net.Security;
+// using Microsoft.AspNetCore.Hosting;
+// using Microsoft.AspNetCore.Server.Kestrel.Core;
+// using Microsoft.Extensions.DependencyInjection;
+// using Microsoft.Extensions.Hosting;
 
-.ConfigureKestrel(serverOptions =>
+webBuilder.ConfigureKestrel(serverOptions =>
 {
-    serverOptions.Listen(IPAddress.Any, 8000, listenOptions =>
+    serverOptions.ConfigureHttpsDefaults(listenOptions =>
     {
-        listenOptions.UseHttps("testCert.pfx", "testPassword");
-        listenOptions.UseConnectionHandler<TlsFilterConnectionHandler>();
+        listenOptions.OnAuthenticate = (context, sslOptions) =>
+        {
+            sslOptions.CipherSuitesPolicy = new CipherSuitesPolicy(
+                new[]
+                {
+                    TlsCipherSuite.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+                    TlsCipherSuite.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+                    // ...
+                });
+        };
     });
 });
 ```
 
-```csharp
-using System;
-using System.Buffers;
-using System.Security.Authentication;
-using System.Threading.Tasks;
-using Microsoft.AspNetCore.Connections;
-using Microsoft.AspNetCore.Connections.Features;
-
-public class TlsFilterConnectionHandler : ConnectionHandler
-{
-    public override async Task OnConnectedAsync(ConnectionContext connection)
-    {
-        var tlsFeature = connection.Features.Get<ITlsHandshakeFeature>();
-
-        // Throw NotSupportedException for any cipher algorithm that the app doesn't
-        // wish to support. Alternatively, define and compare
-        // ITlsHandshakeFeature.CipherAlgorithm to a list of acceptable cipher
-        // suites.
-        //
-        // A ITlsHandshakeFeature.CipherAlgorithm of CipherAlgorithmType.Null
-        // indicates that no cipher algorithm supported by Kestrel matches the
-        // requested algorithm(s).
-        if (tlsFeature.CipherAlgorithm == CipherAlgorithmType.Null)
-        {
-            throw new NotSupportedException("Prohibited cipher: " + tlsFeature.CipherAlgorithm);
-        }
-
-        while (true)
```

Comment:
In 3.0 this copy loop is not needed regardless of platform
https://github.com/aspnet/AspNetCore/blob/24ec6ec8d3ef30183539313438b474ee984cb166/src/Servers/Kestrel/samples/Http2SampleApp/Program.cs#L41-L55